### PR TITLE
[Chip] Add `<Chip />` component, test, and story

### PR DESCRIPTION
**Pull request by snags88**
This adds the Chip component to Radiance

[Storybook](https://curology-radiance-pr-27.herokuapp.com/?selectedKind=Tag&selectedStory=Usage&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybooks%2Fstorybook-addon-knobs)

### CODE REVIEW

File: src/shared-components/tag/style.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import styled, { css } from 'react-emotion';
+
+import { style as TYPOGRAPHY_STYLE } from '../typography';
+import { COLORS, SPACING } from '../../constants';
+
+const closedStyle = css`
+  background-color: ${COLORS.statusGreyBackground};
+  border-color: ${COLORS.statusGreyBorder};
+  span {
```
Comment (smmullen90) [inline src/shared-components/tag/style.js]: [note] i **think** if you rearrange, you can actually interpolate `TagText` here to make this more specific

Comment (snags88) [reply src/shared-components/tag/style.js]: Yup that makes sense. Will update 👍

**Comment by snags88** [comment]
@smmullen90 I can chat with ben regarding the naming!

File: src/shared-components/tag/style.js (hunk 1 above)
Comment (benkolde) [reply src/shared-components/tag/style.js]: @snags88 @smmullen90 I believe we have `badge` already. 

It's this: 
image

It is similar though, wonder if we could have it as two separate types of `badges`. 

I'll throw it up in the slack channel for people to discuss!

**Comment by snags88** [comment]
Consensus was to go with `chip`

**Comment by snags88** [comment]
[Convert Chip component](https://trello.com/c/sJKwmiji/241-convert-chip-component)

**Pull request merged**